crates/core/src/rpc/full.rs

```diff
@@ -1956,11 +1956,61 @@ impl Full for SurfpoolFullRpc {
 
     fn get_signatures_for_address(
         &self,
-        _meta: Self::Metadata,
-        _address: String,
+        meta: Self::Metadata,
+        address: String,
         _config: Option<RpcSignaturesForAddressConfig>,
     ) -> BoxFuture<Result<Vec<RpcConfirmedTransactionStatusWithSignature>>> {
-        not_implemented_err_async("get_signatures_for_address")
+        let address = match verify_pubkey(&address) {
+            Ok(s) => s,
+            Err(e) => return e.into(),
+        };
+        let SurfnetRpcContext {
+            svm_locker,
+            remote_ctx,
+        } = match meta.get_rpc_context(None) {
+            Ok(res) => res,
+            Err(e) => return e.into(),
+        };
+
+        Box::pin(async move {
+            let sigs: Vec<_> = svm_locker.with_svm_reader(|svm_reader| {
+                svm_reader
+                    .transactions
+                    .iter()
+                    .filter(|(_, status)| {
+                        let TransactionWithStatusMeta(_, tx, _, _) = status.expect_processed();
+
+                        tx.message
+                            .static_account_keys()
+                            .iter()
+                            .find_position(|pubkey| pubkey.to_bytes() == address.to_bytes())
+                            .map(|(idx, _)| tx.message.is_signer(idx))
+                            .unwrap_or_default()
+                    })
+                    .map(|(sig, status)| (sig.to_owned(), status.to_owned()))
+                    .collect()
+            });
```

Comment:
Let's make this a function in the `locker.rs` file on `svm_locker`. On the locker, we'll need to implement:
 - `get_signatures_for_address_local`
 - `get_signatures_for_address_local_then_remote`: gets from local, gets from remote, then merges them
 - `get_signatures_for_address` - routes to either `get_signatures_for_address_local` or `get_signatures_for_address_local_then_remote` depending on if there is a `remote_ctx` or not

This is similar to the `get_accounts` implementation on the locker.

The reason is - the requested address could have some transactions on mainnet _before_ the surfnet was created. The account could also have been involved with new transactions locally. We'll want to include both in this list.